lockapi.pm

```diff
@@ -20,36 +20,66 @@ use strict;
 use warnings;
 
 use base qw/Exporter/;
-our @EXPORT = qw/mutex_create mutex_lock mutex_unlock/;
+our @EXPORT = qw/mutex_create mutex_lock mutex_unlock mutex_try_lock/;
 
 require bmwqemu;
 use mmapi qw/api_call/;
 
-sub mutex_lock($) {
-    my ($name) = @_;
-    return _mutex_call('post', "mutex/$name/lock");
+sub mutex_lock($;$) {
+    my ($name, $where) = @_;
+    bmwqemu::diag("mutex lock '$name'");
+    while (1) {
+        my $param = {action => 'lock'};
+        $param->{where} = $where if $where;
```

Comment:
missing where argument means the default - search self and parents, it is handled on server side